### PR TITLE
Add workaround for bug in older linux kernels handling epoll_wait(...)

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Older linux kernels have problems handling a large value for epoll_wait(...) and so wait for ever.

Modifications:

Adjust timeout on the fly if a too big value is passed in.

Result:

Correctly works also on older kernels.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@trustin please review... thanks!

File: transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.c [hunk 1]
```diff
@@ -32,6 +32,17 @@
 #include <sys/utsname.h>
 #include "io_netty_channel_epoll_Native.h"
 
+
```
Comment (trustin) [inline transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.c]: This newline is not necessary.

File: transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.c [hunk 2]
```diff
@@ -620,11 +631,16 @@ JNIEXPORT jint JNICALL Java_io_netty_channel_epoll_Native_epollCreate(JNIEnv* en
...
     int len = (*env)->GetArrayLength(env, events);
     struct epoll_event ev[len];
     int ready;
     int err;
+
+    if (timeout > MAX_EPOLL_TIMEOUT_MSEC) {
+        // Workaround for bug in older linux kernels that can not handle bigger timeout then MAX_EPOLL_TIMEOUT_MSEC.
+        timeout = MAX_EPOLL_TIMEOUT_MSEC;
+    }
     do {
```
Comment (trustin) [inline transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.c]: Add a new line between `if` and `do` block for readability.

**Comment by trustin** [comment]
Looks good to me. Please cherry-pick after addressing the trivial comments.

**Comment by normanmaurer** [comment]
Cherry-picked into 4.0 (795de8a5901da57920fdc64771117773c25b14d3) , 4.1 (0a452bef8edbf023fd950eb6d3875acaea553cdb) and master (340287f377e48159b91d60f6bec84dcc98316192)

**Pull request closed without merging**